### PR TITLE
storage: fix quiescent replica metric

### PR DESCRIPTION
**Pull request by petermattis**
Broken by #12026

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/12180)

### CODE REVIEW

**Review by a-robinson — APPROVED** [review]
LGTM

File: pkg/storage/replica_test.go [hunk 1]
```diff
@@ -7088,15 +7088,16 @@ func TestReplicaMetrics(t *testing.T) {
 				underreplicated: false,
 			}},
 	}
-	for _, c := range testCases {
+	for i, c := range testCases {
 		t.Run("", func(t *testing.T) {
 			zoneConfig := config.ZoneConfig{NumReplicas: c.replicas}
 			defer config.TestingSetDefaultZoneConfig(zoneConfig)()
 
+			c.expected.quiescent = i%2 == 0
```
Comment (tamird) [inline pkg/storage/replica_test.go]: this is sure to confuse future readers. is it really worth the line savings?

Comment (petermattis) [inline pkg/storage/replica_test.go]: What would you prefer?

**Comment by tamird** [comment]
Manually setting this field in the test cases above.

**Comment by petermattis** [comment]
> Manually setting this field in the test cases above.

That requires updating every test case which doesn't seem useful. I've added a comment to make it clearer what is being done.

**Pull request merged**